### PR TITLE
FEAT: Get users based on overdue tasks

### PR DESCRIPTION
**Pull request by sahsisunny**
**Description:**

This pull request introduces a new feature to the user list API, allowing users to filter the list based on overdue tasks within a specified time frame. The API endpoint is as follows:

**Endpoint:** `GET /users?query=filterBy:overdue_tasks+days:<days>`

**Feature Highlights:**

1. **Filtering by Overdue Tasks:** Users can now include a query parameter, `filterBy:overdue_tasks+days:X`, where `X` represents the number of days. This query parameter filters the user list to include only those users who have overdue tasks within the specified time frame.

2. **Improved User Experience:** This feature enhances the user experience by providing a convenient way to identify users with pending tasks that are overdue, helping teams manage their workload efficiently.

3. **Efficient Task Management:** By easily identifying users with overdue tasks, project managers and team leads can take timely actions to address task delays and ensure project milestones are met.

**Usage Example:**

To retrieve a list of users who have overdue tasks within the last 1 day, you can make a GET request to the following endpoint:

**Example:** `GET /users?query=filterBy:overdue_tasks+days:5`

**Response:**

```json
{
  {
    "message": "Users returned successfully!",
    "count": 3,
    "users": [
        "NNKUYC49Q7vpK8aD7K44",
        "ZsgadltDqexIpsSeR9DH",
        "WbceXEPcohuJ5IxHHecf"
        ]
    }
}
```

**Testing Coverage:**
- Unit Test
image

- Integration Test
image

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -128,6 +128,20 @@ const getUsers = async (req, res) => {
...
+        return res.json({
+          message: "Users returned successfully!",
+          count: users.length,
+          users: users,
+        });
+      } catch (error) {
+        logger.error(`Error while fetching all users: ${error}`);
+        return res.boom.serverUnavailable("Something went wrong please contact admin");
+      }
+    }
```
Comment (Achintya-Chatterjee) [inline controllers/users.js:143]: error handling logic seems appropriate. You catch errors that may occur during the execution of await userQuery.getUserswithOverdueTasks(days), log the error, and return a serverUnavailable response. Just ensure that the logger object is correctly configured in your application to log errors effectively.

**Review by shreya-mishra — DISMISSED** [review]
LGTM

File: test/unit/models/users.test.js [hunk 2]
```diff
@@ -455,4 +456,35 @@ describe("users", function () {
...
+      await cleanDb();
+    });
+    it("should return the users which have overdue tasks", async function () {
+      const days = 10;
+      const usersWithOverdueTasks = await users.getUserswithOverdueTasks(days);
+      expect(usersWithOverdueTasks.length).to.be.equal(3);
+      expect(usersWithOverdueTasks).to.include("ankur");
+      expect(usersWithOverdueTasks).to.include("akshay");
+      expect(usersWithOverdueTasks).to.include("narendra");
+      expect(usersWithOverdueTasks).to.not.include("modi");
```
Comment (prakashchoudhary07) [inline test/unit/models/users.test.js:480]: 😆

**Review by Pratiyushkumar — APPROVED** [review]

**Review by fakhruddinkw — APPROVED** [review]
LGTM

**Review by Pratiyushkumar — APPROVED** [review]

**Review by Achintya-Chatterjee — APPROVED** [review]

**Pull request merged**